pyluna-pathology/luna/pathology/cli/generate_tile_mask.py

```diff
@@ -0,0 +1,113 @@
+
+
+logger = logging.getLogger("convert_tiles_to_mask") 
+init_logger()
+
+_params_ = [('input_slide_image', str), ('input_slide_tiles', str), ('output_dir', str), ('label_cols', List[str])]
+
+@click.command()
+@click.argument('input_slide_image', nargs=1)
+@click.argument('input_slide_tiles', nargs=1)
+@click.option('-o', '--output_dir', required=False,
+              help='path to output directory to save results')
+@click.option('-lc', '--label_cols', required=False, 
+              help="columns whose values are used to generate the mask")
+@click.option('-m', '--method_param_path', required=False, 
+              help='path to a metadata json/yaml file with method parameters to reproduce results')
+
+def cli(**cli_kwargs):
+    """ Generates a .tif mask from slide tile labels
+
+    \b
+    Inputs:
+        input: input data
+    \b
+    Outputs:
+        output data
+    \b
+    Example:
+        convert_tiles_to_mask ./slides/10001.svs ./tiles_scores_and_labels.csv
+            -lc Background,Tumor
+            -o ./label_mask.csv
+    """
+    cli_runner( cli_kwargs, _params_, convert_tiles_to_mask)
+
+
+def convert_tiles_to_mask(input_slide_image:str, input_slide_tiles:str, label_cols:List[str], output_dir:str):
+    """ Converts cateogrial tile labels to a slide image mask. This mask can be used for feature extraction and spatial analysis. 
+
+     Args:
+        input_slide_image (str): path to slide image (virtual slide formats compatible with openslide, .svs, .tif, .scn, ...)
+        input_slide_tiles (str): path to valid SlideTiles table
+        label_cols (List[str]): list of label columns in the input_slide_tiles table to generate the mask with
+        output_dir (str): output/working directory
+
+    Returns:
+        dict: output .tif path and the mask size 
+
+    """
+
+    slide = openslide.OpenSlide(input_slide_image)
+
+    w  = slide.dimensions[0]
+    h = slide.dimensions[1] 
+    wsi_shape = h, w # open slide has reversed conventions
+    logger.info(f"Slide shape={wsi_shape}")
+
+    logger.info(f"Validating SlideTiles schema")
+    assert SlideTiles.check(f"{input_slide_tiles}") # verfiying SlideTiles schema
+
+    # check if tile_col is a valid argument
+    logger.info(f"Reading SlideTiles")
+    print(label_cols)
+    tile_df = pd.read_csv(f"{input_slide_tiles}")
+    if not set(label_cols).issubset(tile_df.columns):
+        raise ValueError(f"Invalid label_cols={label_cols}, verify input dataframe")
+        
+    mask_arr = np.zeros((h, w), dtype=np.int8) 
+
+    tile_df['mask'] = tile_df[label_cols].idxmax(axis=1)
+
+    mask_values = {k: v+1 for v, k in enumerate(label_cols)}
+    logger.info(f"Mapping label column to mask values: {mask_values}")
+
+    for _, row in tile_df.iterrows():
+        address = row['address']
+        
+        tile_size = int(row['tile_size'])
+        x, y, z = address.split('_')
+        x = int(x.strip('x')) - 1
+        y = int(y.strip('y')) - 1   
```

Comment:
Eventually the x/y_coord attribute should be used here, I think the coordinates needs to be fixed though.

Reply:
Agreed. yeah, the coordinates are offset by the tile size I'm pretty sure, so they refer to the bottom right corner as opposed to the top left.